src/network/networkd-dhcp4.c

```diff
@@ -649,10 +649,23 @@ static int dhcp4_handler(sd_dhcp_client *client, int event, void *userdata) {
                                         return log_link_warning_errno(link, r, "Could not acquire IPv4 link-local address: %m");
                         }
 
+                        if (link->network->dhcp_critical) {
+                                log_link_notice(link, "DHCPv4 connection considered system critical, ignoring request to reconfigure it.");
```

Comment:
"... is considered critical, ignoring ..." (There is no "system" in the name or description of CriticalConnection=.)